File: management-domain/Customer-support/Customer-support/Backend/Node.js/chatbot-service/src/shared/context/request-context.ts
export class RequestContext {
  private static _current: RequestContext | null = null;

  constructor(
    public readonly requestId: string,
    public readonly userId?: string,
    public readonly tenantId?: string,
    public readonly role?: string,
  ) {}

  static set current(ctx: RequestContext | null) {
    RequestContext._current = ctx;
  }

  static get current(): RequestContext | null {
    return RequestContext._current;
  }
}

File: management-domain/Sales-department/Backend/Java/notification-service/src/main/java/com/gogidix/sales/notification/shared/requestcontext/RequestContext.java
package com.gogidix.sales.notification.shared.requestcontext;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Map;

/**
 * Request Context
 * Contains contextual information about the current request
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class RequestContext {

    private String tenantId;

    private String userId;

    private String correlationId;

    private String requestId;

    private String userAgent;

    private String ipAddress;

    private String locale;

    private Map<String, Object> metadata;
}
